File: Strivers A2Z DSA Sheet/217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 8/7/2026, 7:13:34 PM
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (int i = 0; i < nums.size(); i++) {
            if (seen.count(nums[i])) {
                return true;
            }
            seen.insert(nums[i]);
        }
        return false;
    }
};

File: Strivers A2Z DSA Sheet/217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 8/7/2026, 7:13:29 PM
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> list = new HashSet();
        for(int i:nums){
            if(list.add(i) == false) return true;
        }
        return false;
    }
}